mods/ra/maps/soviet-10/soviet10.lua

```diff
@@ -0,0 +1,349 @@
+ConvoyTrucks = { "truk", "truk", "truk" }
+LightVehicleAttack = { Jeep1, Jeep2, Ltnk }
+LightVehicleAttackTrigger = { CPos.New(94, 55), CPos.New(94, 56), CPos.New(94, 57), CPos.New(94, 58), CPos.New(94, 59), CPos.New(94, 70), CPos.New(94, 71), CPos.New(94, 72), CPos.New(94, 73) }
+WinTriggerArea = { CPos.New(16, 55), CPos.New(16, 56), CPos.New(16, 57), CPos.New(16, 58), CPos.New(16, 59), CPos.New(16, 60), CPos.New(16, 61), CPos.New(16, 62), CPos.New(16, 63), CPos.New(16, 64), CPos.New(16, 65), CPos.New(16, 66) }
+EngiDropTriggerArea = { CPos.New(66, 51), CPos.New(66, 52), CPos.New(66, 53), CPos.New(66, 54), CPos.New(66, 55), CPos.New(66, 67), CPos.New(66, 68), CPos.New(66, 69), CPos.New(66, 70) }
+ChinookAmbushTrigger = { CPos.New(54, 70), CPos.New(55, 70), CPos.New(56, 70), CPos.New(57, 70), CPos.New(58, 70), CPos.New(59, 70), CPos.New(60, 70), CPos.New(61, 70) }
+ChinookChalk = { "e3", "e3", "e3", "e3", "e3" }
+ChinookPath = { ChinookEntry.Location, ChinookLZ.Location }
+PatrolA = { "e1", "e1", "e1", "e1", "e1" }
+PatrolB = { "e1", "e1", "e1", "e3", "e3" }
+PatrolAPath = { PatrolA1.Location, PatrolA2.Location, PatrolA3.Location, PatrolA4.Location }
+PatrolBPath = { PatrolB1.Location, PatrolB2.Location, PatrolB3.Location, PatrolB4.Location }
+HunterTeam = { "1tnk", "1tnk", "1tnk", "1tnk", "2tnk", "2tnk", "arty" }
+
+StartTimer = false
+TimerColor = Player.GetPlayer("USSR").Color
+TimerTicks = DateTime.Minutes(9)
+ticked = TimerTicks
+StartTimerDelay =
+{
+	easy = DateTime.Minutes(5),
+	normal = DateTime.Minutes(3),
+	hard = DateTime.Minutes(1)
+}
+
+MissionStart = function()
+	Utils.Do(StartingPlanes, function(a)
+		a.ReturnToBase()
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		StartTank.AttackMove(TrucksStop.Location)
+		Reinforcements.Reinforce(USSR, ConvoyEscort, { ConvoyEntry.Location, TanksStop.Location })
+		Media.PlaySpeechNotification(USSR, "ConvoyApproaching")
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		local patrol1 = Reinforcements.Reinforce(Greece, PatrolA, { SouthEntry.Location, PatrolA1.Location })
+		Utils.Do(patrol1, function(unit)
+			Trigger.OnIdle(unit, function(patrols)
+				patrols.Patrol(PatrolAPath, true, 80)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		local patrol2 = Reinforcements.Reinforce(Greece, PatrolB, { NorthEntry.Location, PatrolB1.Location })
+		Utils.Do(patrol2, function(unit)
+			Trigger.OnIdle(unit, function(patrols)
+				patrols.Patrol(PatrolBPath, true, 80)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(8), function()
+		Trucks = Reinforcements.Reinforce(Turkey, ConvoyTrucks, { ConvoyEntry.Location, TrucksStop.Location })
+		TrucksFollow()
+		Trigger.OnAnyKilled(Trucks, function()
+			USSR.MarkFailedObjective(ProtectEveryTruck)
+		end)
+	end)
+end
+
+MissionSetup = function()
+	Trigger.OnEnteredFootprint(WinTriggerArea, function(a, id)
+		if a.Type == "truk" then
+			--stop truck from looping through TrucksFollow
```

Comment:
Style nit: Comments should have a space and start with a capital letter (`-- Stop truck...`)